spring-boot/src/main/java/org/springframework/boot/web/support/ErrorPageFilter.java

```diff
@@ -223,12 +223,12 @@ private String getErrorPath(Class<?> type) {
 		if (this.subtypes.containsKey(type)) {
 			return this.exceptions.get(this.subtypes.get(type));
 		}
-		Class<?> subtype = type;
-		while (subtype != Object.class) {
-			subtype = subtype.getSuperclass();
-			if (this.exceptions.containsKey(subtype)) {
-				this.subtypes.put(subtype, type);
-				return this.exceptions.get(subtype);
+		Class<?> supertype = type;
+		while (supertype != Object.class) {
+			supertype = supertype.getSuperclass();
+			if (this.exceptions.containsKey(supertype)) {
+				this.subtypes.put(type, supertype);
```

Comment:
This looks like more than a typo correction. You've swapped the key and value in this call to `put`. Was that intentional?